### ISSUE TITLE
[BUG] plot_prediction_actual_by_variable mixes up actual/prediction for time_varying_known_categoricals

### ISSUE THREAD

**Issue by mkuiack** [sktime/pytorch-forecasting#1788]
**Describe the bug**

I've noticed that the plot labels under the `plot_prediction_actual_by_variable` `time_varying_known_categoricals` do not correspond to the correct axis label. I made a simple example where a value increases  each day throughout the week then verified that the labeled actuals are not correct. 

**To Reproduce**

I define a simple time-series, make a TFT prediction then evaluate the actual vs prediction. 

Actual vs prediction plot:
![Image](https://github.com/user-attachments/assets/6aa828db-1f0b-4bd1-821f-ce0b1c7be246)
Actual actuals: 
![Image](https://github.com/user-attachments/assets/6aefe9f3-4c6a-4ffc-b3b2-c70f4bf0355c)

```python
# Define dataset
max_encoder_length = 30 * 24  # Use past 30 days * 24 hours
max_prediction_length = 7 * 24  # Predict the next 7 days

training_cutoff = (
    df["time_idx"].max() - max_prediction_length
)  # Last part reserved for validation

training = TimeSeriesDataSet(
    df[lambda x: x.time_idx <= training_cutoff],
    time_idx="time_idx",
    target="total_sales",  # Multi-target: ["orders_per_hour", "total_sales",]
    group_ids=["group_id"],  # Single time series
    min_encoder_length=max_encoder_length,
    max_encoder_length=max_encoder_length,
    min_prediction_length=max_prediction_length,
    max_prediction_length=max_prediction_length,
    static_categoricals=[],
    time_varying_known_categoricals=["day_name", "month_name"],
    time_varying_known_reals=[
        "hour",
        "day_of_week",
    ],
    time_varying_unknown_reals=[
        "orders_per_hour",
        "total_sales",
    ],
    target_normalizer=GroupNormalizer(groups=[], transformation="softplus"),
)

validation = TimeSeriesDataSet.from_dataset(
    training, df, predict=True, stop_randomization=True
)

# Convert to PyTorch dataloaders
batch_size = min(32, len(df) // (max_encoder_length + max_prediction_length))

train_dataloader = training.to_dataloader(
    train=True, batch_size=batch_size, num_workers=0
)
val_dataloader = validation.to_dataloader(
    train=False, batch_size=batch_size, num_workers=0
)

# Define TFT model
tft = TemporalFusionTransformer.from_dataset(
    training,
    learning_rate=0.001,
    hidden_size=16,
    attention_head_size=4,
    dropout=0.1,
    hidden_continuous_size=8,
    output_size=1,
    loss=MAE(),
    log_interval=10,
    reduce_on_plateau_patience=4,
    optimizer="adam",
)

# Train TFT model
trainer = Trainer(max_epochs=1, accelerator="cpu")
trainer.fit(tft, train_dataloaders=train_dataloader, val_dataloaders=val_dataloader)

predictions = tft.predict(
    val_dataloader, return_x=True, trainer_kwargs=dict(accelerator="cpu")
)
predictions_vs_actuals = tft.calculate_prediction_actual_by_variable(
    predictions.x, predictions.output
)
tft.plot_prediction_actual_by_variable(predictions_vs_actuals, name="day_name")
```

**Expected behavior**

The labels in the plot should follow the days of the week: Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday 

**Versions**
<details>

pytorch_forecasting = "1.3.0"

</details>

**Comment by mkuiack** [comment]
The issue is that `plot_prediction_actual_by_variable` assumes `self.hparams.embedding_labels` to be in the same order as the values in `averages_actual_cat` but that's not the case. 

Does anyone know why this _should_ be the case? 

![Image](https://github.com/user-attachments/assets/27b1c5c1-7950-4fc6-ad28-5ee799130c77)
